src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -541,7 +548,42 @@ SHAMapStoreImp::clearSql (DatabaseCon& database,
             std::this_thread::sleep_for (
                     std::chrono::milliseconds (setup_.backOff));
     }
-    journal_.debug << "finished: " << deleteQuery;
+    JLOG(journal_.debug) << "finished: " << deleteQuery;
+}
+
+void
+SHAMapStoreImp::clearSql(DatabaseCon& database,
+    std::string const& deleteQuery)
+{
+    auto const deleteBatch = setup_.deleteBatch;
+    auto const continueLimit = (deleteBatch + 1) / 2;
+    auto const formattedDeleteQuery =
+        boost::str(boost::format(deleteQuery) % deleteBatch);
+
+    JLOG(journal_.debug) <<
+        "start: " << deleteQuery << " of " << deleteBatch << " rows.";
+    long long totalRowsAffected = 0;
+    long long rowsAffected;
+    do
+    {
+        {
+            auto db = database.checkoutDb();
+            soci::statement st = (db->prepare << formattedDeleteQuery);
```

Comment:
Note: I would still lock the database when executing the statement. I'm not 100% sure it's needed, but until I can prove otherwise I'd keep the lock.